lib/resources/docker.rb

```diff
@@ -0,0 +1,341 @@
+# encoding: utf-8
+#
+# Copyright 2017, Christoph Hartmann
+#
+# author: Christoph Hartmann
+# author: Patrick Muench
+# author: Dominik Richter
+
+require 'utils/filter'
+require 'hashie/mash'
+
+class DockerContainerFilter
+  # use filtertable for containers
+  filter = FilterTable.create
+  filter.add_accessor(:where)
+        .add_accessor(:entries)
+        .add(:commands,       field: 'command')
+        .add(:ids,            field: 'id')
+        .add(:images,         field: 'image')
+        .add(:labels,         field: 'labels')
+        .add(:local_volumes,  field: 'localvolumes')
+        .add(:mounts,         field: 'mounts')
+        .add(:names,          field: 'names')
+        .add(:networks,       field: 'networks')
+        .add(:ports,          field: 'ports')
+        .add(:running_for,    field: 'runningfor')
+        .add(:sizes,          field: 'size')
+        .add(:status,         field: 'status')
+        .add(:exists?) { |x| !x.entries.empty? }
+  filter.connect(self, :containers)
+
+  attr_reader(:containers)
```

Comment:
Can we remove the `( )` here? It's customary to not include them when declaring attributes (i.e. `attr_reader :containers`)